### ISSUE TITLE
Support shared properties of type unions

### ISSUE THREAD

**Issue by sonygod** [haxiomic/dts2hx#105]
ts.AnyOf issue.

some gen class have some fields like 

```haxe
    	var geometry : ts.AnyOf2<A,B>;

        geometry .AProperty=1;//current will throw error.
      
       geometry.BProperty=2;//current will throw error.
```

//is possible to do something like that?

**Comment by haxiomic** [comment]
I’m not sure I follow - could you explain a bit more? What does the source typescript look like?

Do you mean those properties are shared between types A and B and should be accessible within the AnyOf?

I have a plan for this but it requires macros, it’s in the roadmap but less critical than other features

**Comment by sonygod** [comment]
Do you mean those properties are shared between types A and B and should be accessible within the AnyOf?

yes ,I mean  those properties are shared between types A and B,because we have a project use native js ,there are lots of these

style something like 
      
```
         class  SomeClass {
           var geometry : ts.AnyOf2<A,B>;
     }

        var sc=new SomeClass ();

       sc.geometry .AProperty=1;
   
       var sc2=new SomeClass ();

      sc.geometry .BProperty=1;
```

         ....and so on

   I convert to haxe I have to write lots of  case ugly code ...

**Comment by haxiomic** [comment]
Agreed, it can be solved with a type building macro which is on the roadmap
I'll rename the issue and leave it open

btw this also works
```haxe
sc.asType0.AProperty = 1;
sc.asType1.BProperty = 1;
```

**Comment by sonygod** [comment]
there is more complex in three.js  

```
              class  SomeClass {
           var geometry : ts.AnyOf2<AParent,BParent>;
     }
      
        var sc=new SomeClass ();

       sc.geometry .AChildClasssProperty=1;//native js  direct support  child's property.
```

**Comment by sonygod** [comment]
@haxiomic ,dts2hx 0.18.1 seem support this ? I will close this issue.
